src/main/java/org/templegalaxia/patterns/acjs/Spoken.java

```diff
@@ -0,0 +1,216 @@
+              .setDescription("Length of a 'dot' in morse code.");
+
+  public final CompoundParameter dotVariance =
+      (CompoundParameter)
+          new CompoundParameter("Dot variance", 0, 0, 1000)
+              .setUnits(LXParameter.Units.MILLISECONDS)
+              .setDescription("How variable dot length will be between names.");
+
+  private final HashMap<Character, String> morseLookup = new HashMap<>();
+  private final List<String> names = new ArrayList<>();
+  private final List<MorsePulser> pulsers = new ArrayList<>();
+  private final List<Integer> availableLights = new ArrayList<>();
+
+  private final Random rand = new Random();
+
+  public Spoken(LX lx) {
+    super(lx);
+    addParameter(density);
+    addParameter(birthRate);
+    addParameter(dot);
+    addParameter(dotVariance);
+
+    loadMorse();
+
+    for (int i = 0; i < colors.length; i++) availableLights.add(i);
+  }
+
+  void loadMorse() {
+    // Shamelessly boosted from the model load code.
+    CSVParser parser;
+    try {
+      parser =
+          new CSVParser(
+              new BufferedReader(
+                  new InputStreamReader(Spoken.class.getResourceAsStream("morse.csv"))),
+              CSVFormat.EXCEL.withHeader());
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+
+    for (CSVRecord csvRecord : parser) {
+      morseLookup.put(csvRecord.get("Letter").charAt(0), csvRecord.get("Morse"));
+    }
+  }
+
+  void loadNames() {
+    Path dataPath = Paths.get(NAME_DATA);
+
+    if (Files.exists(dataPath) && Files.isRegularFile(dataPath)) {
+      try (Stream<String> nameLines = Files.lines(dataPath)) {
+        nameLines.forEach(names::add);
+      } catch (IOException e) {
+        e.printStackTrace();
+      } finally {
+        // Backup / default
+        names.add("Larry Harvey");
+      }
+    }
+
+    Collections.shuffle(names);
+  }
+
+  public void newPulser(String message) {
+    double dotLength = dot.getValue();
+    dotLength += (Math.random() - 0.5) * dotVariance.getValue();
+
+    Collections.shuffle(availableLights);
+    int light = availableLights.remove(0);
+
+    MorsePulser pulser = new MorsePulser(lx, light, message, dotLength);
+    pulsers.add(pulser);
+    addLayer(pulser);
+  }
+
+  public void run(double deltaMs) {
+    // Give each unlit 'slot' a chance to be born.
+    for (int i = 0; i < (int) density.getValue() - pulsers.size(); i++) {
+      if (names.size() == 0) loadNames(); // Reload names if we're out. XXX: Performance?
+
+      if (rand.nextDouble() < birthRate.getValue() * deltaMs / 1000) {
+        newPulser(names.remove(0));
+      }
+    }
+
+    // Garbage collect completed pulsers
+    // TODO: Do this with a callback on finish or something.
+    for (Iterator<MorsePulser> iterator = pulsers.iterator(); iterator.hasNext(); ) {
+      MorsePulser pulse = iterator.next();
+      if (pulse.isFinished()) {
+        availableLights.add(pulse.getLightIndex());
+        iterator.remove();
```

Comment:
I think you also want to removeLayer() here. Otherwise the layer still stays on the LXPattern and its run() method will be called every frame.